File: server/internal/models/race.go
package models

import "time"

type Race struct {
	Year       int       `json:"year"`
	Round      int       `json:"round"`
	RaceId     string    `json:"raceId"`
	RaceName   string    `json:"raceName"`
	Circuit    Circuit   `json:"circuit"`
	Date       time.Time `json:"date"`
	CutoffTime time.Time `json:"cutoffTime"`
}

type Circuit struct {
	CircuitId   string   `json:"circuitId"`
	URL         string   `json:"url"`
	CircuitName string   `json:"circuitName"`
	Location    Location `json:"location"`
}

type Location struct {
	Latitude  float32 `json:"lat"`
	Longitude float32 `json:"lng"`
	Country   string `json:"country"`
}

File: client/formula1-fantasy/models/race.ts
export interface Race {
    year: number;
    round: number;
    raceId: string;
    raceName: string;
    circuit: Circuit;
    date: string;
    cutoffTime: string;
    imagePath: string;
}

interface Circuit {
    circuitId: string;
    url: string;
    circuitName: string;
    location: Location;
}

interface Location {
    lat: number;
    lng: number;
    country: string;
}